### PR TITLE
NAT-213: load office details into database

### PR DESCRIPTION
**Pull request by cnorthwood**
This PR adds the following:
* sets up a database and data structure for holding local Citizens Advice office details
* adds a class that can load in LSS data files into that database

### CODE REVIEW

File: lib/lss_loader.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      validate_csv_headers!
+      ActiveRecord::Base.connection.truncate(Office.table_name)
+      save_offices_by_tier! build_office_records
+    end
+  end
+
+  private
+
+  def validate_csv_headers!
+    raise LssLoadError, "Accounts CSV file was not in expected format" unless @account_csv.headers == %w[
```
Comment (KPobeeNorris) [inline lib/lss_loader.rb]: The number of headers is quite a lot to read in this method so I wondered if they could be extracted out into their own method along the lines of `account_csv_headers_valid?`?  This is just a personal preference in terms of making the overall method a bit easier to read

File: spec/models/office_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Office do
+  it "can round trip to the database a minimal example" do
```
Comment (KPobeeNorris) [inline spec/models/office_spec.rb]: I admit I've read this `it` statement a few times and can't quite figure out what it means 😬

File: db/migrate/20230531135320_create_offices.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        execute <<~SQL
+          DROP TYPE timerange;
+          DROP FUNCTION time_subtype_diff;
+        SQL
+      end
+    end
+
+    create_enum :office_type, %w[member office outreach]
+
+    # rubocop:disable Rails/CreateTableWithTimestamps
```
Comment (KPobeeNorris) [inline db/migrate/20230531135320_create_offices.rb:32]: For my own learning - is there a reason this table doesn't have timestamps?

**Review by KPobeeNorris — APPROVED** [review]
Lots of new stuff I'd not seen before in this PR 😄 

There's a couple of comments, and a question for my own learning, but happy to approve 🦖

File: spec/models/office_spec.rb (hunk 2 above)
Comment (cnorthwood) [reply spec/models/office_spec.rb]: I've rephrased this one in the most recent commit to hopefully be clearer in intent!

File: db/migrate/20230531135320_create_offices.rb (hunk 3 above)
Comment (cnorthwood) [reply db/migrate/20230531135320_create_offices.rb:32]: I added a comment explaining the rationale (short answer is that they're not needed and I'm worried they would confuse!)

**Review by KPobeeNorris — APPROVED** [review]
Thanks for making those changes 😃

**Pull request merged**